### DISCUSSION TITLE
Toggle single attribute multiple times

### DISCUSSION THREAD

**Discussion by karlomikus** [algolia/instantsearch#5416] [Q&A]
I have a multiple known list of attribute values, for example:
- Favorites: [1, 2, 3, 4]
- My shelf: [90, 105]
- My first list: [3, 4, 5, 6, 7, 8, 9]

These values are "static" and I want to have a list of refinements that include them (Favorites, My Shelf, My First list).

Currently I can use toggle refinement and give it an array of values:

``` vue
// Toggle favorites
<ais-toggle-refinement attribute="id" :on="[1, 2, 3, 4]"></ais-toggle-refinement>
// Toggle other
<ais-toggle-refinement attribute="id" :on="[3, 4, 5, 6, 7]"></ais-toggle-refinement>
```

This works but has some issues:
- Routing is always added as: `http://url.com/resources?toggle[id]=true` so you can't know what refinement is currently active
- The `current-refinements` component can't distinct between 2 different refinements since they use the same attribute

Here's an example with multiple toggles: https://codesandbox.io/s/eloquent-brook-hr9ohz?file=/src/App.vue

**Comment by Haroenv** [comment]
This could indeed cause some edge-cases. I am currently not sure what fits this use case the best. 

The most simple solution would likely be to make a small patch to `connectToggleRefinement` in your project to allow it to accept a `identifier` value too, which should be used instead of attribute in getWidgetUiState and getWidgetSearchParameters. Then the URL no longer overlaps.

For current-refinements you could filter in transformItems and turn that attribute in your "identifier"

**Comment by karlomikus** [comment]
Great, I think I get it.
I copied the connector and added `identifier` param that will be my "label" for current refinements and routing.

I updated `getWidgetUiState` with the following

``` ts
getWidgetUiState(uiState, { searchParameters }) {
  const isRefined =
    on &&
    on.every((v) =>
      searchParameters.isDisjunctiveFacetRefined(attribute, v)
    );

  if (!isRefined) {
    return uiState;
  }

  return {
    ...uiState,
    toggle: {
      ...uiState.toggle,
      [identifier]: isRefined,
    },
  };
},
```

And the following function:
``` ts
getWidgetSearchParameters(searchParameters, { uiState }) {
  let withFacetConfiguration = searchParameters
    .clearRefinements(attribute)
    .addDisjunctiveFacet(attribute);

  const isRefined = Boolean(
    uiState.toggle && uiState.toggle[identifier]
  );

  if (isRefined) {
    if (on) {
      on.forEach((v) => {
        withFacetConfiguration =
          withFacetConfiguration.addDisjunctiveFacetRefinement(
            attribute,
            v
          );
      });
    }

    return withFacetConfiguration;
  }

  // It's not refined with an `off` value
  if (hasAnOffValue) {
    if (off) {
      off.forEach((v) => {
        withFacetConfiguration =
          withFacetConfiguration.addDisjunctiveFacetRefinement(
            attribute,
            v
          );
      });
    }
    return withFacetConfiguration;
  }

  // It's not refined without an `off` value
  return withFacetConfiguration.setQueryParameters({
    disjunctiveFacetsRefinements: {
      ...searchParameters.disjunctiveFacetsRefinements,
      [identifier]: [],
    },
  });
},
```

Now when I select refinement I get correctly updated url, ex: `toggle[favorites]=true`. But when I manually visit the url like `https://url.com?toggle[favorites]=true` the refinement disappears. I guess I'm missing something in `getWidgetUiState` method.
